ibazel/ibazel_test.go

```diff
@@ -256,7 +252,7 @@ func TestIBazelLoop(t *testing.T) {
 	assertRun()
 	assertState(WAIT)
 	// Source file change.
-	i.sourceEventHandler.SourceFileEvents <- fsnotify.Event{Op: fsnotify.Write, Name: "/path/to/foo"}
+	go func() { i.sourceFileWatcher.Events() <- common.Event{Op: common.Write, Name: "/path/to/foo"} }()
```

Comment:
I'm really sorry to be throwing one last thing out before we cross the finish line, but this looks very wrong to me. In this test, the state machine is single threaded with blocks in order to ensure there are no race conditions. I presume you put a state transition in a go routine because it was blocking. That means that there is an unhandled state transition, or race condition in the actual code introduced in this PR.

I went through and did a quick audit of the changes you made and they all look okay. Could you try removing the `go` I think it might just work (TM). If it doesn't can you tell me and I'll look into it with you.

Reply:
Yea, this was an interesting one. In my refactoring, I remove "[source_event_handler](https://github.com/bazelbuild/bazel-watcher/pull/406/files#diff-58ef3a9da67919bd663ea42bf5d6b4b02516d72b70815129d241e69d135619fc).go" (git thinks it was changed, but i deleted it). In that file, there was a goroutine that watched and replayed events on the source file channel. As that was no longer needed in my refactoring, I had to put the go routine here instead or else it blocks forever waiting either for the channel sender or the first value to come through. I am very new to go, so I could be missing something else, but that is my understanding. If I remove this go func here, it hangs and times-out.

Reply:
```exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //ibazel:go_default_test
-----------------------------------------------------------------------------
=== RUN   TestIBazelLifecycle
[96miBazel [9:12PM][0m: You are using an experimental filesystem watcher. If you would like to disable that, please set the environment variable
	IBAZEL_USE_LEGACY_WATCHER=1
--- PASS: TestIBazelLifecycle (0.00s)
[96miBazel [9:12PM][0m: Querying for files to watch...
=== RUN   TestIBazelLoop
Error finding remote repositories directory: open external: no such file or directory
Error finding remote repositories directory: open external: no such file or directory
[96miBazel [9:12PM][0m: Demoing 
-- Test timed out at 2020-12-01 21:17:44 UTC --
```

Reply:
Doesn't that mean that the order of operations for the state machine should have changed and we need to update this test to reflect the new state of the world?

Reply:
I think the state machine operates in the same order..?
Could you explain more please?

Reply:
If the state machine hasn't changed, then why would this test start failing? Something involved with the state machine must have changed, either the machine it self or the way that the inputs respond. Either way we will have to change this test, I just want to make sure that I understand what changes are being made

Reply:
Ah ok. Yes I think before my change, the state machine was not single threaded, see the previously linked file that I removed. After my change, the state machine is single threaded but required the tests to not be single theaded.

Reply:
Any ideas?

Reply:
Sorry, I'm in a work crunch. Can I get back to you on Monday?

Reply:
Ping! ;) Have you had any spare time for this lately @achew22?